### PR TITLE
add basic webview component

### PR DESCRIPTION
**Pull request by ssseasonnn**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## In the scope
add basic webview component

## Checklist
1. [x] Makes your submission pass tests?
2. [x] Have you lint your code locally before submission?
3. [ ] Have you updated the wiki of your change?

## Reference
- Card: https://trello.com/c/BfL5WRfF/111-h5-container

### CODE REVIEW

File: webview/src/main/java/com/thoughtworks/ark/webview/WebViewClientImpl.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class WebViewClientImpl(
+    val updateTitle: (String) -> Unit,
+    val updateErrorState: (Boolean) -> Unit,
+    val shouldOverrideUrl: (String) -> Boolean = { false }
+) : WebViewClient() {
+
+    override fun onReceivedError(view: WebView?, request: WebResourceRequest?, error: WebResourceError?) {
+        super.onReceivedError(view, request, error)
+        updateErrorState(true)
```
Comment (Tony---Zhang) [inline webview/src/main/java/com/thoughtworks/ark/webview/WebViewClientImpl.kt]: 这个Error是不是要判断一下Error Code啊？都按照404处理吗？

File: webview/src/main/java/com/thoughtworks/ark/webview/WebViewTitleBar.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import android.widget.TextView
+
+class WebViewTitleBar @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : FrameLayout(context, attrs) {
+    private var onBackClick: () -> Unit = {}
+    private var onCloseClick: () -> Unit = {}
+
+    private val btnBack: ImageView
```
Comment (Tony---Zhang) [inline webview/src/main/java/com/thoughtworks/ark/webview/WebViewTitleBar.kt]: 不接受btn跟tv这种无意义的缩写命名，请修改

File: webview/src/main/res/layout/web_view_title_bar.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="?android:attr/actionBarSize"
+    android:background="?attr/colorPrimary">
+
+    <ImageView
+        android:id="@+id/btn_back"
```
Comment (Tony---Zhang) [inline webview/src/main/res/layout/web_view_title_bar.xml]: 请修改id名称

File: webview/src/main/java/com/thoughtworks/ark/webview/WebViewTitleBar.kt (hunk 2 above)
Comment (ssseasonnn) [reply webview/src/main/java/com/thoughtworks/ark/webview/WebViewTitleBar.kt]: fixed

File: webview/src/main/res/layout/web_view_title_bar.xml (hunk 3 above)
Comment (ssseasonnn) [reply webview/src/main/res/layout/web_view_title_bar.xml]: fixed

File: webview/src/main/java/com/thoughtworks/ark/webview/WebViewClientImpl.kt (hunk 1 above)
Comment (ssseasonnn) [reply webview/src/main/java/com/thoughtworks/ark/webview/WebViewClientImpl.kt]: fixed

**Review by Tony---Zhang — APPROVED** [review]

**Review by weiraneve — APPROVED** [review]

**Pull request merged**